torch/csrc/jit/symbolic_script.cpp

```diff
@@ -6,20 +6,253 @@ namespace {
 std::mutex lock;
 const std::vector<std::string> functions = {
     R"(
+
+        def _dim_arange(like,
+                        dim: int):
+            def backward(grad_output):
+                return None, None
+
+            return torch._dim_arange(like, dim), backward
+
+        def contiguous(self):
+            def backward(grad_output):
+                return None
+
+            return self.contiguous(), backward
+
+        def erf(self):
+            def backward(grad_output):
+                M_PI = 3.14159265358979323846
+                grad_self = 2.0 / torch.sqrt(torch.tensor(M_PI)) * torch.exp(- self.pow(2)) * grad_output
+                return grad_self
+
+            return torch.erf(self), backward
+
+        def expand(self,
+                   size: List[int],
+                   implicit: bool=False):
+            def backward(grad_output):
+                grad_self = torch._grad_sum_to_size(grad_output, self.size())
+                return grad_self, None, None
+
+            return torch.expand(self, size, implicit=implicit), backward
+
+        def expand_as(self, other):
+            def backward(grad_output):
+                grad_self = grad_output._grad_sum_to_size(self.size())
+                return grad_self, None
+
+            return torch.expand_as(self, other), backward
+
+        def full_like(self,
+                      fill_value: float):
+            def backward(grad_output):
+                return None, None
+
+            return torch.full_like(self, fill_value), backward
+
+        def kthvalue(self,
+                     k: int,
+                     dim: int,
+                     keepdim: bool):
+            result0, result1 = torch.kthvalue(self, k, dim, keepdim)
+            def backward(grad_output):
+                grad_self = torch.index_select_backward(grad_output, dim, result1, self.size(), keepdim)
+                return grad_self, None, None, None
+
+            return result0, result1, backward
+
+        def mean_0(self):
+            def backward(grad_output):
+                grad_self = grad_output.expand(self.size()) / self.numel()
+                return grad_self
+
+            return torch.mean(self), backward
+
+        def mean_1(self,
```

Comment:
What is `mean_0` and `mean_1`? How do we match up those things with definitions later?

Reply:
These suffixes will be removed later when looking up function schema cache. This is a hack to simulate function overloading we have in aten. https://github.com/pytorch/pytorch/pull/16689/files/1a3d72ec52ddc4755008dfe920d4c0a25b8a1e13#diff-758cb565d2c84fc5268de0b63958f78dR330

Reply:
- name: mean(Tensor self)




- name: mean(Tensor self, ScalarType dtype)
- name: mean(Tensor self, IntList dim, bool keepdim)
- name: mean(Tensor self, IntList dim, ScalarType dtype)
- name: mean(Tensor self, IntList dim, bool keepdim, ScalarType dtype)

There are mean overloads that has dtype in the signature, should we also include those overloads?

Reply:
@wanchaol Yea but this `dtype` option is not exposed in our public api. Thus these ops are not specifically tested in common_methods_invocations.py. Do you know some ops that call into the dtype version of mean? 
I didn't want to add some ops without proper tests, thus those were not in this PR.

Reply:
I found one callsites in [here](https://github.com/pytorch/pytorch/blob/41dddfd55fc59a0ebae7cda7e759b80011741a78/test/test_cuda.py#L2017), but that's only a test and no ops found that calls `mean` with dtype, so I guess it should be fine.

Reply:
aha this option is actually exposed but not documented :( `a = torch.mean(a, dtype=torch.double)` works. I'm can add AD & test for them.

Reply:
Actually I realized all dtypes here are keyword only arguments, and since we don't document them I assume it's rarely used in real models :P 
I would prefer to add them together with proper keyword-only argument parser so that we are less hacky on this. But if there's a usecase that needs this AD formula, let me know so that I can prioritize. :D

Reply:
I don't think we have too much use cases on this either. Let's just add this with the keyword arg parser together.